app/models/job.ts

```diff
@@ -399,14 +395,20 @@ export class Job extends Record {
     this.validateStatus();
     this.message = truncateString(this.message, 4096);
     this.request = truncateString(this.request, 4096);
-    // Need to jump through serialization hoops due array caveat here: http://knexjs.org/#Schema-json
     const { links, originalStatus } = this;
     delete this.links;
     delete this.originalStatus;
-    this._json_links = JSON.stringify(links);
-    await super.save(transaction);
+    // Remove the _json_links field once we've dropped the column
+    await super.save(transaction, { ...this, _json_links: [] });
+    const promises = [];
+    for (const link of links) {
```

Comment:
This will probably need to change to be a bit more O(1) once we get into pagination and no granule limit, i.e. we saving the link and attaching it to the job is probably going to need to happen on the link object rather than the link array in the job object.